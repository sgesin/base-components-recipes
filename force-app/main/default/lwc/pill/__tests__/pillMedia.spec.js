/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import Element from 'lightningtest/pillTest';

const createComponent = () => {
    const element = createElement('lightningtest-pill-test', { is: Element });
    document.body.appendChild(element);
    return element;
};

describe('pill with media elements', () => {
    it('label', () => {
        const element = createComponent();
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
