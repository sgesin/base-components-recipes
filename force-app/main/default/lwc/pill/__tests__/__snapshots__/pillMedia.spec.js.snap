// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pill with media elements label 1`] = `
<lightningtest-pill-test>
  #shadow-root(open)
    <c-pill
      title="Title for Pill"
    >
      #shadow-root(open)
        <span
          class="slds-pill slds-pill_link"
        >
          <span
            class="slds-pill__icon_container"
          >
            <slot />
          </span>
          <a
            class="slds-pill__action"
            href="/path/to/somewhere"
            title="Pill with Avatar"
          >
            <span
              class="slds-pill__label"
            >
              Pill with Avatar
            </span>
          </a>
          <c-button-icon
            class="slds-pill__remove"
            tabindex="0"
          >
            #shadow-root(open)
              <button
                class="slds-button slds-button_icon slds-button_icon-bare"
                title="Remove"
                type="button"
              >
                <c-primitive-icon>
                  #shadow-root(open)
                    <svg
                      aria-hidden="true"
                      class="slds-button__icon"
                      data-key="close"
                      focusable="false"
                    >
                      <use
                        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#close"
                      />
                    </svg>
                </c-primitive-icon>
                <span
                  class="slds-assistive-text"
                >
                  Remove
                </span>
              </button>
          </c-button-icon>
        </span>
      <c-avatar
        class="slds-avatar slds-avatar_medium slds-avatar_circle"
      >
        #shadow-root(open)
          <img
            alt="User avatar"
            src="/assets/images/avatar2.jpg"
            title="User avatar"
          />
      </c-avatar>
    </c-pill>
    <c-pill
      title="Title for Pill"
    >
      #shadow-root(open)
        <span
          class="slds-pill slds-pill_link"
        >
          <span
            class="slds-pill__icon_container"
          >
            <slot />
          </span>
          <a
            class="slds-pill__action"
            href="/path/to/somewhere"
            title="Pill with Icon"
          >
            <span
              class="slds-pill__label"
            >
              Pill with Icon
            </span>
          </a>
          <c-button-icon
            class="slds-pill__remove"
            tabindex="0"
          >
            #shadow-root(open)
              <button
                class="slds-button slds-button_icon slds-button_icon-bare"
                title="Remove"
                type="button"
              >
                <c-primitive-icon>
                  #shadow-root(open)
                    <svg
                      aria-hidden="true"
                      class="slds-button__icon"
                      data-key="close"
                      focusable="false"
                    >
                      <use
                        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#close"
                      />
                    </svg>
                </c-primitive-icon>
                <span
                  class="slds-assistive-text"
                >
                  Remove
                </span>
              </button>
          </c-button-icon>
        </span>
      <c-icon
        class="slds-icon-standard-account slds-icon_container"
      >
        #shadow-root(open)
          <c-primitive-icon>
            #shadow-root(open)
              <svg
                aria-hidden="true"
                class="slds-icon"
                data-key="account"
                focusable="false"
              >
                <use
                  xlink:href="/assets/icons/standard-sprite/svg/symbols.svg#account"
                />
              </svg>
          </c-primitive-icon>
          <span
            class="slds-assistive-text"
          >
            Account
          </span>
      </c-icon>
    </c-pill>
    <c-pill
      title="Title for Pill"
    >
      #shadow-root(open)
        <span
          class="slds-pill"
        >
          <span
            class="slds-pill__icon_container"
          >
            <slot />
          </span>
          <span
            class="slds-pill__label"
            title="Pill with Avatar"
          >
            Pill with Avatar
          </span>
          <c-button-icon
            class="slds-pill__remove"
            tabindex="0"
          >
            #shadow-root(open)
              <button
                class="slds-button slds-button_icon slds-button_icon-bare"
                title="Remove"
                type="button"
              >
                <c-primitive-icon>
                  #shadow-root(open)
                    <svg
                      aria-hidden="true"
                      class="slds-button__icon"
                      data-key="close"
                      focusable="false"
                    >
                      <use
                        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#close"
                      />
                    </svg>
                </c-primitive-icon>
                <span
                  class="slds-assistive-text"
                >
                  Press delete or backspace to remove
                </span>
              </button>
          </c-button-icon>
        </span>
      <c-avatar
        class="slds-avatar slds-avatar_medium slds-avatar_circle"
      >
        #shadow-root(open)
          <img
            alt="User avatar"
            src="/assets/images/avatar2.jpg"
            title="User avatar"
          />
      </c-avatar>
    </c-pill>
    <c-pill
      title="Title for Pill"
    >
      #shadow-root(open)
        <span
          class="slds-pill"
        >
          <span
            class="slds-pill__icon_container"
          >
            <slot />
          </span>
          <span
            class="slds-pill__label"
            title="Pill with Icon"
          >
            Pill with Icon
          </span>
          <c-button-icon
            class="slds-pill__remove"
            tabindex="0"
          >
            #shadow-root(open)
              <button
                class="slds-button slds-button_icon slds-button_icon-bare"
                title="Remove"
                type="button"
              >
                <c-primitive-icon>
                  #shadow-root(open)
                    <svg
                      aria-hidden="true"
                      class="slds-button__icon"
                      data-key="close"
                      focusable="false"
                    >
                      <use
                        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#close"
                      />
                    </svg>
                </c-primitive-icon>
                <span
                  class="slds-assistive-text"
                >
                  Press delete or backspace to remove
                </span>
              </button>
          </c-button-icon>
        </span>
      <c-icon
        class="slds-icon-standard-account slds-icon_container"
      >
        #shadow-root(open)
          <c-primitive-icon>
            #shadow-root(open)
              <svg
                aria-hidden="true"
                class="slds-icon"
                data-key="account"
                focusable="false"
              >
                <use
                  xlink:href="/assets/icons/standard-sprite/svg/symbols.svg#account"
                />
              </svg>
          </c-primitive-icon>
          <span
            class="slds-assistive-text"
          >
            Account
          </span>
      </c-icon>
    </c-pill>
    <c-pill
      title="Title for pill"
    >
      #shadow-root(open)
        <span
          class="slds-pill"
        >
          <span
            class="slds-pill__label"
            title="Pill with standard container class name"
          >
            Pill with standard container class name
          </span>
          <c-primitive-icon
            class="slds-icon_container slds-pill__remove"
            title="Remove"
          >
            #shadow-root(open)
              <svg
                aria-hidden="true"
                class="slds-icon slds-icon_x-small slds-icon-text-default"
                data-key="close"
                focusable="false"
              >
                <use
                  xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#close"
                />
              </svg>
          </c-primitive-icon>
          <span
            class="slds-assistive-text"
          >
            Press delete or backspace to remove
          </span>
        </span>
    </c-pill>
    <c-pill
      title="Title for pill"
    >
      #shadow-root(open)
        <span
          class="slds-pill slds-pill_link"
        >
          <a
            class="slds-pill__action"
            href="/path/to/somewhere"
            title="Pill with standard container class name"
          >
            <span
              class="slds-pill__label"
            >
              Pill with standard container class name
            </span>
            <c-primitive-icon
              class="slds-icon_container slds-pill__remove"
              title="Remove"
            >
              #shadow-root(open)
                <svg
                  aria-hidden="true"
                  class="slds-icon slds-icon_x-small slds-icon-text-default"
                  data-key="close"
                  focusable="false"
                >
                  <use
                    xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#close"
                  />
                </svg>
            </c-primitive-icon>
            <span
              class="slds-assistive-text"
            >
              Press delete or backspace to remove, press enter to navigate
            </span>
          </a>
        </span>
    </c-pill>
</lightningtest-pill-test>
`;
